### PR TITLE
Use pointycastle as crypto library. cipher is deprecated.

### PR DESCRIPTION
**Pull request by mpcomplete**

### CODE REVIEW

File: packages/flx/pubspec.yaml [hunk 1]
```diff
@@ -6,7 +6,7 @@ homepage: https://github.com/flutter/flutter/tree/master/packages/flx
 dependencies:
   bignum: ^0.0.7
   asn1lib: ^0.4.1
-  cipher: ^0.7.1
+  pointycastle: ^0.10.0-dev
```
Comment (devoncarew) [inline packages/flx/pubspec.yaml]: We want just `0.10.0`. `-dev` is good if you need to consume a pre-release package.

Comment (mpcomplete) [inline packages/flx/pubspec.yaml]: I don't think there is an 0.10.0 yet:

> pub get
Resolving dependencies... 
Package pointycastle has no versions that match >=0.10.0 <0.11.0 derived from:
- flx depends on version ^0.10.0

Comment (devoncarew) [inline packages/flx/pubspec.yaml]: I see them published here: https://pub.dartlang.org/packages/pointycastle, so I think what you're hitting it something in another constraint forcing an older version. `pub` has a mode you can run it in that will give you more info about why it is or isn't selecting a version. Might be worth doing that to diagnose -

Comment (mpcomplete) [inline packages/flx/pubspec.yaml]: Yep! You're right. I switched to a newer version of bignum, and it works now. Thanks.

**Comment by mpcomplete** [comment]
Still struggling with `flutter analyze` on this one. It's hanging with this patch. Even without this patch, running it locally gives me a bunch of warnings. Investigating.

**Comment by eseidelGoogle** [comment]
You can try running dartanalyzer directly.  You might be hitting an analyzer bug in which case @pq is your man.

**Comment by pq** [comment]
> You can try running dartanalyzer directly. You might be hitting an analyzer bug in which case @pq is your man.

:+1:  Let me know if you hit any snags.

**Comment by mpcomplete** [comment]
I think it's dying on some lint checks with one of my transitive dependencies. This command hangs:

cd packages/flx && dartanalyzer --package-warnings --lints lib/signing.dart

**Comment by pq** [comment]
Great.  I've got it reproed here.  Thanks!

**Comment by pq** [comment]
Hmmm, it looks like there's an issue with the `LibraryNames` lint.  In particular, we're borking when testing the library named `pointycastle.impl.ec_domain_parameters.gostr3410_2001_cryptopro_a` using an (admittedly gnarly) regexp.  Digging into `RegExp`...

**Comment by pq** [comment]
First thought for a work-around: the analyzer defaults to a set of lints that are thought to be uncontroversial and (errr) safe.  To exclude the suspect `library_names` we could specify the rules explicitly using an `.analysis_options` file.  In particular, adding this to `flutter/packages/flx` seems to do the trick.

```
linter:
  rules:
    - camel_case_types
    - constant_identifier_names
    - empty_constructor_bodies
    # - library_names
    - library_prefixes
    - non_constant_identifier_names
    - one_member_abstracts
    - slash_for_doc_comments
    - super_goes_last
    - type_init_formals
    - unnecessary_brace_in_string_interp
```

For example, I see:

```
> dartanalyzer --package-warnings --lints lib/signing.dart
  102 hints and 924 lints found.
```

(What you're going to do with those 924 lints is another question!)

**Comment by pq** [comment]
OK, regexp cleanup is in flight too: https://codereview.chromium.org/1719553002/.

**Comment by abarth** [comment]
Hi, this pull request is assigned to me.  Is there some action you'd like me to take on it?

**Comment by pq** [comment]
FWIW the linter fix will be in the next dev build.  It's landed in the SDK (https://github.com/dart-lang/sdk/commit/2e7baf61be0a3817b23e52b4c49b4c858ef16782) and is in the latest linter available on pub (`0.1.12`).

**Comment by mpcomplete** [comment]
Reassigning to Devon since he already reviewed it. I'm going to have to wait for us to get the new version of the dart SDK, though.

**Comment by pq** [comment]
A short-term fix (and where we want to go eventually) may be to update our `_embedder.yaml` to exclude the `library_names`.  Investigating...

**Comment by devoncarew** [comment]
Yup, I think we'll need to disable the lint (in the current stable SDK the regex that lint uses doesn't terminate), and then ignore the other lints that come from the pointycastle package. That will probably mean updating our list of uber analysis regexs (https://github.com/flutter/flutter/blob/master/packages/flutter_tools/lib/src/commands/analyze.dart#L290).

**Comment by pq** [comment]
The fix in https://github.com/flutter/flutter/commit/b69d249460c404561c453417a5556c0d005eff59 should unblock this.  Let me know if not!

**Comment by devoncarew** [comment]
I patched this in locally and updated to head; it `flutter analyze`s clean. lgtm

**Pull request merged**